python/ray/_private/structured_logging/logging_config.py

```diff
@@ -0,0 +1,52 @@
+from typing import Optional
+
+from ray._private.structured_logging.constants import LOG_MODE_DICT
+from ray.util.annotations import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class LoggingConfig:
+    def __init__(self, encoding: Optional[str] = "TEXT", log_level: str = "INFO"):
+        """
+        The class is used to store the Python logging configuration. It will be applied
+        to the root loggers of the driver process and all Ray task and actor processes
+        that belong to the this job.
+
+        Examples:
+            .. testcode::
+
+                import ray
+                import logging
+
+                ray.init(
+                    logging_config=ray.LoggingConfig(encoding="TEXT", log_level="INFO")
+                )
+
+                @ray.remote
+                def f():
+                    logger = logging.getLogger()
+                    logger.info("This is a Ray task")
+
+                obj_ref = f.remote()
+                ray.get(obj_ref)
+
```

Comment:
Let's add

```
.. testoutput::
   :options: +MOCK

    2024-06-02 22:06:57,539	INFO <stdin>:1 -- This is a Ray task job_id=01000000 worker_id=01000000ffffffffffffffffffffffffffffffffffffffffffffffff node_id=ea6aa650b81f193d8929dcc680ab6dc290ea45c16bf893bef005f37
```

to showcase the log output

Reply:
Updated https://github.com/ray-project/ray/pull/45344/commits/229f9d47bb26ef204cd8380e75d092bc587e90b1